### PR TITLE
Fix RMT on esp32s3 (and other variants).

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: lib/rmt.toit [hunk 1]
```diff
@@ -232,6 +232,10 @@ class Channel:
   /**
   Constructs a channel using the given $num using the given $pin.
 
+  Note: only the ESP32 and the ESP32S2 support configuring the channel direction at a later
+    time. For all other platforms this constructor will give a TX channel, unless
```
Comment (kasperl) [inline lib/rmt.toit:236]:
```suggestion
    time. For all other platforms, this constructor will give a TX channel, unless
```

File: src/resources/rmt_esp32.cc [hunk 2]
```diff
@@ -31,17 +31,74 @@ namespace toit {
...
 #ifndef SOC_RMT_CHANNELS_PER_GROUP
 #error "SOC_RMT_CHANNELS_PER_GROUP not defined"
 #endif
 
-ResourcePool<rmt_channel_t, kInvalidChannel> rmt_channels(
-    RMT_CHANNEL_0, RMT_CHANNEL_1, RMT_CHANNEL_2, RMT_CHANNEL_3
+#if SOC_RMT_CHANNELS_PER_GROUP == SOC_RMT_TX_CANDIDATES_PER_GROUP
+// All channels are TX channels. We assume that they also are RX channels.
+// This is the setup on the ESP32 and ESP32S2 chips.
+ResourcePool<rmt_channel_t, kInvalidChannel> rmt_channels_(
```
Comment (kasperl) [inline src/resources/rmt_esp32.cc:49]: Should these be static (pools and pointers)?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: lib/rmt.toit (hunk 1 above)
Comment (floitsch) [reply lib/rmt.toit]: done.

File: src/resources/rmt_esp32.cc (hunk 2 above)
Comment (floitsch) [reply src/resources/rmt_esp32.cc]: done.

**Pull request merged**